backend/de.metas.business/src/main/java/de/metas/tax/interceptor/C_Tax.java

```diff
@@ -0,0 +1,110 @@
+import de.metas.location.ICountryDAO;
+import de.metas.organization.OrgId;
+import de.metas.tax.api.TypeOfDestCountry;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.modelvalidator.annotations.Interceptor;
+import org.adempiere.ad.modelvalidator.annotations.ModelChange;
+import org.compiere.model.I_C_Tax;
+import org.compiere.model.ModelValidator;
+import org.compiere.util.Env;
+import org.springframework.stereotype.Component;
+
+import static de.metas.tax.api.TypeOfDestCountry.DOMESTIC;
+import static de.metas.tax.api.TypeOfDestCountry.OUTSIDE_COUNTRY_AREA;
+import static de.metas.tax.api.TypeOfDestCountry.WITHIN_COUNTRY_AREA;
+
+@Interceptor(I_C_Tax.class)
+@Component
+public class C_Tax
+{
+	private final ICountryAreaBL countryAreaBL = Services.get(ICountryAreaBL.class);
+	private final ICountryDAO countryDAO = Services.get(ICountryDAO.class);
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE }, ifColumnsChanged = { I_C_Tax.COLUMNNAME_To_Country_ID })
+	public void updateTypeOfDestCountry(@NonNull final I_C_Tax tax)
+	{
+		final CountryId toCountryId = CountryId.ofRepoIdOrNull(tax.getTo_Country_ID());
+		if (toCountryId == null)
+		{
+			final OrgId orgId = OrgId.ofRepoId(tax.getAD_Org_ID());
+			if (orgId.isAny())
+			{
+				tax.setTypeOfDestCountry(WITHIN_COUNTRY_AREA.getCode());
+			}
+		}
+		else
+		{
+			if (tax.getC_Country_ID() == tax.getTo_Country_ID())
+			{
+				tax.setTypeOfDestCountry(DOMESTIC.getCode());
+			}
+			else
+			{
+				final String countryCode = countryDAO.retrieveCountryCode2ByCountryId(toCountryId);
+				final boolean isEULocation = countryAreaBL.isMemberOf(Env.getCtx(),
+						ICountryAreaBL.COUNTRYAREAKEY_EU,
+						countryCode,
+						tax.getUpdated());
+				final TypeOfDestCountry typeOfDestCountry = isEULocation ? WITHIN_COUNTRY_AREA : OUTSIDE_COUNTRY_AREA;
+				tax.setTypeOfDestCountry(typeOfDestCountry.getCode());
+
+			}
+		}
+	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE }, ifColumnsChanged = { I_C_Tax.COLUMNNAME_TypeOfDestCountry })
+	public void updateToCountryId(@NonNull final I_C_Tax tax)
+	{
+		final TypeOfDestCountry typeOfDestCountry = TypeOfDestCountry.ofNullableCode(tax.getTypeOfDestCountry());
+		final int countryId = tax.getC_Country_ID();
+		final int toCountryIdRepoId = tax.getTo_Country_ID();
+		if ((DOMESTIC.equals(typeOfDestCountry) && countryId != toCountryIdRepoId) || (countryId == toCountryIdRepoId && !DOMESTIC.equals(typeOfDestCountry)))
+		{
+			tax.setTo_Country_ID(0);
+		}
+		else
+		{
+			final CountryId toCountryId = CountryId.ofRepoIdOrNull(toCountryIdRepoId);
+			if (toCountryId != null)
+			{
+				final String countryCode = countryDAO.retrieveCountryCode2ByCountryId(toCountryId);
+				final boolean isEULocation = countryAreaBL.isMemberOf(Env.getCtx(),
+						ICountryAreaBL.COUNTRYAREAKEY_EU,
+						countryCode,
+						tax.getUpdated());
```

Comment:
not sure which value `tax.getUpdated()` has. maybe better use `SystemTime`

Reply:
fixed.